packages/sdk/src/chains/Substrate/types/index.ts

```diff
@@ -1,8 +1,21 @@
+import { AccountInfo } from '@polkadot/types/interfaces';
+import { ethers } from 'ethers';
+import { FeeHandlerType } from '../../../types';
+
 export type XcmMultiAssetIdType = {
   concrete: {
     parents: number;
     interior: {
-      x3: Array<{ parachain: number } | { generalKey: string }>;
+      x3: Array<{ parachain: number } | { generalKey: [number, string] }>; // This is a tuple of length and value
     };
   };
 };
+
+export type SubstrateFee = {
+  fee: ethers.BigNumber;
+  type: FeeHandlerType;
+  tokenAddress?: string;
```

Comment:
Is this relevant. At this point fee data and token address are both nkt usefull as we don't have dynamic fee yet.